src/app/content/components/ContentLink.tsx

```diff
@@ -14,17 +16,18 @@ interface Props extends React.HTMLProps<HTMLAnchorElement> {
     title: string;
   };
   navigate: typeof push;
+  currentPath: string;
   className?: string;
 }
 
 // tslint:disable-next-line:variable-name
-export const ContentLink: SFC<Props> = ({book, page, navigate, ...props}) => {
+export const ContentLink: SFC<Props> = ({book, page, currentPath, navigate, ...props}) => {
   const params = {
     book: book.slug,
     page: getUrlParamForPageId(book, page.shortId),
   };
 
-  const url = content.getUrl(params);
+  const url = relative(dirname(currentPath), content.getUrl(params)) || `../${basename(params.page)}`;
```

Comment:
i assume this `||` is to catch the case where `If from and to each resolve to the same path (after calling path.resolve() on each), a zero-length string is returned.` but i don't see any test cases for the requested and current path being the same

Reply:
also i think it might be technically correct to leave such a case as an empty href....

Reply:
The case occurs when the current url is something like `/books/A/pages/B/C`.

In this case we could maybe get away with `""` but:

- I'm not sure if browser handle that case properly
- this would mean that the target URL would be `/books/A/pages/B/` rather than `/books/A/pages/B` (no trailing slash) and I don't know if amazon handles that

Reply:
interesting... does it also happen when the requested and current path are the same? i don't see a test for that case.

also this is inconsistent with the "deeply nested" one that just returns ".."

Reply:
I think I addressed the first concern in the recent commit. The 2nd concern is unused... fixing it would likely add more code... should we handle it?